lms/djangoapps/course_api/serializers.py

```diff
@@ -0,0 +1,125 @@
+"""
+Course API Serializers.  Representing course catalog data
+"""
+
+import urllib
+
+from django.core.urlresolvers import reverse
+from django.template import defaultfilters
+
+from rest_framework import serializers
+
+from xmodule.course_module import DEFAULT_START_DATE
+from lms.djangoapps.courseware.courses import course_image_url, get_course_about_section
+
+
+class _MediaSerializer(serializers.Serializer):  # pylint: disable=abstract-method
+    """
+    Nested serializer to represent a media object.
+    """
+    def __init__(self, name=u'', description=u'', type_=u'', uri_parser=lambda x: x, *args, **kwargs):
+        super(_MediaSerializer, self).__init__(*args, **kwargs)
+
+        self.name_value = name
+        self.description_value = description
+        self.type_value = type_
+        self.uri_parser = uri_parser
+
+    uri = serializers.SerializerMethodField(source='*')
+    name = serializers.SerializerMethodField()
+    description = serializers.SerializerMethodField()
+    type = serializers.SerializerMethodField()  # pylint: disable=redefined-builtin
+
+    def get_name(self, _):
+        """
+        Get the representation for the media resource's name
+        """
+        return self.name_value
+
+    def get_description(self, _):
+        """
+        Get the representation for the media resource's description
+        """
+        return self.description_value
+
+    def get_type(self, _):
+        """
+        Get the representation for the media resource's type
+        """
+        return self.type_value
+
+    def get_uri(self, course):
+        """
+        Get the representation for the media resource's URI
+        """
+        return self.uri_parser(course)
+
+
+class _CourseApiMediaCollectionSerializer(serializers.Serializer):  # pylint: disable=abstract-method
+    """
+    Nested serializer to represent a collection of media objects
+    """
+    course_image = _MediaSerializer(
+        source='*',
```

Comment:
Because otherwise, it looks for a field named "course_image" on the object that gets passed to it.  The * makes it pass in the whole object instead. (remove the source="*" and you get lots of errors.)